shared/contracts/BaseTemplate.sol

```diff
@@ -56,155 +62,178 @@ contract BaseTemplate is APMNamehash, IsContract {
-        acl.createPermission(address(this), app, permission, address(this));
+    function _createPermissions(ACL _acl, address[] _grantees, address _app, bytes32 _permission, address _manager) internal {
+        _acl.createPermission(_grantees[0], _app, _permission, address(this));
+        for (uint256 i = 1; i < _grantees.length; i++) {
+            _acl.grantPermission(_grantees[i], _app, _permission);
+        }
+        _acl.revokePermission(address(this), _app, _permission);
+        _acl.setPermissionManager(_manager, _app, _permission);
+    }
+
+    function _createPermissionForTemplate(ACL _acl, address _app, bytes32 _permission) internal {
+        _acl.createPermission(address(this), _app, _permission, address(this));
     }
 
-    function _createPermissions(ACL acl, address[] grantees, address app, bytes32 permission, address manager) internal {
-        acl.createPermission(grantees[0], app, permission, address(this));
-        for (uint256 i = 1; i < grantees.length; i++) {
-            acl.grantPermission(grantees[i], app, permission);
-        }
-        acl.revokePermission(address(this), app, permission);
-        acl.setPermissionManager(manager, app, permission);
+    function _removePermissionFromTemplate(ACL _acl, address _app, bytes32 _permission) internal {
+        _acl.revokePermission(address(this), _app, _permission);
+        _acl.removePermissionManager(_app, _permission);
     }
 
-    function _removePermission(ACL acl, address app, bytes32 permission) internal {
-        acl.revokePermission(address(this), app, permission);
-        acl.removePermissionManager(app, permission);
+    function _transferPermissionFromTemplate(ACL _acl, address _to, address _app, bytes32 _permission) internal {
+        _transferPermissionFromTemplate(_acl, _to, _to, _app, _permission);
     }
 
-    function _transferPermission(ACL acl, address to, address app, bytes32 permission) internal {
-        _transferPermission(acl, to, to, app, permission);
+    function _transferPermissionFromTemplate(ACL _acl, address _to, address _manager, address _app, bytes32 _permission) internal {
+        _acl.grantPermission(_to, _app, _permission);
+        _acl.revokePermission(address(this), _app, _permission);
+        _acl.setPermissionManager(_manager, _app, _permission);
     }
 
-    function _transferPermission(ACL acl, address to, address manager, address app, bytes32 permission) internal {
-        acl.grantPermission(to, app, permission);
-        acl.revokePermission(address(this), app, permission);
-        acl.setPermissionManager(manager, app, permission);
+    function _transferRootPermissionsFromTemplate(Kernel _dao, address _to) internal {
+        ACL _acl = ACL(_dao.acl());
+        _transferPermissionFromTemplate(_acl, _to, _dao, _dao.APP_MANAGER_ROLE());
+        _transferPermissionFromTemplate(_acl, _to, _acl, _acl.CREATE_PERMISSIONS_ROLE());
```

Comment:
I'm not sure we would ever use this version when transferring root permissions; setting the Kernel and ACL as the manager of the roles is basically setting them to `BURN_ROLE`.

Reply:
Not sure what you mean but it is actually being used by the `BareTemplate` to gran root permissions to a root address passed by the user.

Reply:
But I think it uses the other one, that includes the `manager` parameter, right? This one sets the managers to the ACL and Kernel.

Reply:
Sorry, I meant that this one is being used by all the templates except the `BareTemplate`... This helper sets the `_to` address as the manager, note that it is calling the following function for both roles:

```sol
_transferPermissionFromTemplate(ACL _acl, address _to, address _app, bytes32 _permission)
```

However, I'll reuse the other helper used by the `BareTemplate` to make it clearer